api/src/infrastructure/docker.rs

```diff
@@ -286,30 +316,40 @@ impl DockerInfrastructure {
 
     async fn stop_services_impl(&self, app_name: &AppName) -> Result<Vec<Service>, Error> {
         let container_details = match self
-            .get_container_details(Some(app_name), None)
+            .get_container_details(
+                Some(app_name),
+                None,
+                Some(ContainerStateStatusEnum::RUNNING),
+            )
             .await?
             .get_vec(app_name)
         {
             None => return Ok(vec![]),
             Some(services) => services.clone(),
         };
 
-        let futures = container_details
-            .iter()
-            .filter(|details| details.state.running)
-            .map(|details| stop(details.clone()));
-        for container in join_all(futures).await {
-            trace!("Stopped container {:?}", container?);
-        }
+        let docker = Docker::connect_with_socket_defaults()?;
 
-        let mut services = Vec::with_capacity(container_details.len());
-        let futures = container_details
-            .iter()
-            .map(|details| delete(details.clone()));
-        for container in join_all(futures).await {
-            let container = container?;
-            trace!("Deleted container {:?}", container);
+        let mut futures = container_details
+            .into_iter()
+            .map(|details| async {
+                let id = details
+                    .id
+                    .clone()
+                    .expect("id is mandatory for a docker container");
+                docker.stop_container(&id, None).await?;
```

Comment:
Previously, there was filtering on the state `.filter(|details| details.state.running)` before stopping (a container might be stopped in the UI before deleting the app).

Reply:
At line 320-324, I only do a fetch of containers that are running. Anyways I will confirm this with a manual test.
```rust
let container_details = match self
            .get_container_details(
                Some(app_name),
                None,
                Some(ContainerStateStatusEnum::RUNNING),
            )
            .await?
```

Reply:
then you won't delete containers that have been “paused” in the UI.